### PR TITLE
Remove one colon check for well-formed edge tags

### PR DESCRIPTION
**Pull request by juliannzhou**
### What does this PR do?

This PR removes the one colon check for well-formed edge tags.

### Motivation

https://dd.slack.com/archives/C06P5GB4UT1/p1717608298448609

### Reviewer's Checklist

- [ ] Changed code has unit tests for its functionality at or near 100% coverage.
- [ ] [System-Tests](https://github.com/DataDog/system-tests/) covering this feature have been added and enabled with the va.b.c-dev version tag.
- [ ] There is a benchmark for any new code, or changes to existing code.
- [ ] If this interacts with the agent in a new way, a system test has been added.
- [ ] Add an appropriate team label so this PR gets put in the right place for the release notes.
- [ ] Non-trivial go.mod changes, e.g. adding new modules, are reviewed by @DataDog/dd-trace-go-guild.

Unsure? Have a question? Request a review!

### CODE REVIEW

**Comment by darccio** [comment]
@piochelepiotr Is it safe to release the previuosly merged PR or I should wait for this PR?

**Comment by piochelepiotr** [comment]
> @piochelepiotr Is it safe to release the previuosly merged PR or I should wait for this PR?



> @piochelepiotr Is it safe to release the previuosly merged PR or I should wait for this PR?

It's safe to release the previously merged PR.

File: internal/datastreams/pathway_test.go [hunk 1]
```diff
@@ -135,9 +135,9 @@ func TestPathway(t *testing.T) {
 			{"dog:bark", false},
 			{"type:", true},
 			{"type:dog", true},
-			{"type::dog", false},
-			{"type:d:o:g", false},
-			{"type::", false},
+			{"type::dog", true},
```
Comment (piochelepiotr) [inline internal/datastreams/pathway_test.go:138]: Let's add a test for a sns arn, and make sure it passes.

**Review by piochelepiotr — APPROVED** [review]

**Review by darccio — APPROVED** [review]
LGTM

**Pull request merged**